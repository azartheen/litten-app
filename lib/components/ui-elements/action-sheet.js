/**
 * @format
 * @flow strict-local
 */

import { ActionSheetIOS, Alert, Platform } from 'react-native'
import { translate } from 'utils/i18n'

const UIActionSheet = (options, callback) => {
  if (Platform.OS === 'ios') {
    ActionSheetIOS.showActionSheetWithOptions(options, callback)
  } else {
    Alert.alert(translate('feedback.errorMessages.notImplemented'))
  }
}

export default UIActionSheet
