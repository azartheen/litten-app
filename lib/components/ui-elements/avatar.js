/**
 * @format
 * @flow
 */

import React from 'react'
import { Image, StyleSheet, View } from 'react-native'
import { USER_AVATAR_SIZE_MINI } from 'utils/constants'

const UIAvatar: (args: any) => React$Node = ({
  containerStyle,
  size,
  source,
  ...otherProps
}) => {
  const getSize = () => {
    switch (size) {
      default:
        return styles.uiAvatarSizeMini
    }
  }

  const getSource = () => {
    if (typeof source === 'object' && source.uri) {
      return {
        ...source,
        cache: 'force-cache',
      }
    }
    return source
  }

  return (
    <View {...otherProps} style={[containerStyle, getSize()]}>
      {source && (
        <Image
          source={getSource()}
          style={[styles.uiAvatarImage, getSize()]}
          resizeMode="contain"
        />
      )}
    </View>
  )
}

const styles = StyleSheet.create({
  uiAvatarImage: {
    overflow: 'hidden',
  },
  uiAvatarSizeMini: {
    height: USER_AVATAR_SIZE_MINI,
    width: USER_AVATAR_SIZE_MINI,
    borderRadius: USER_AVATAR_SIZE_MINI / 2,
  },
})

export default UIAvatar
