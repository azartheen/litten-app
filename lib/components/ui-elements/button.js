/**
 * @format
 * @flow strict-local
 */

import React from 'react'
import { StyleSheet, Text, TouchableOpacity } from 'react-native'
import { vw } from 'react-native-expo-viewport-units'
import colors from 'styles/colors'

const UIButton: () => React$Node = ({
  children,
  disabled = false,
  onPress,
  secondary,
  style,
  textStyle,
  ...otherProps
}) => (
  <TouchableOpacity
    {...otherProps}
    onPress={disabled ? null : onPress}
    style={StyleSheet.compose(
      secondary
        ? disabled
          ? secondaryDisabledStyle
          : secondaryStyle
        : disabled
        ? primaryDisabledStyle
        : primaryStyle,
      style,
    )}>
    <Text style={StyleSheet.compose(styles.uiButtonText, textStyle)}>
      {children}
    </Text>
  </TouchableOpacity>
)

const styles = StyleSheet.create({
  uiButton: {
    height: 40,
    width: vw(77),
    borderRadius: 20,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: colors.purple,
  },
  uiButtonText: {
    color: colors.white,
    fontSize: 16,
    fontWeight: '600',
  },
  secondary: {
    backgroundColor: colors.blue,
  },
  disabled: {
    opacity: 0.3,
  },
})

const primaryStyle = styles.uiButton
const primaryDisabledStyle = StyleSheet.compose(primaryStyle, styles.disabled)
const secondaryStyle = StyleSheet.compose(primaryStyle, styles.secondary)
const secondaryDisabledStyle = StyleSheet.compose(
  secondaryStyle,
  styles.disabled,
)

export default UIButton
