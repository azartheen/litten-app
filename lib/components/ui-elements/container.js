/**
 * @format
 * @flow
 */

import React from 'react'
import { StyleSheet, View } from 'react-native'

const UIContainer: () => React$Node = ({ children, style, ...otherProps }) => (
  <View {...otherProps} style={StyleSheet.compose(styles.uiContainer, style)}>
    {children}
  </View>
)

const styles = StyleSheet.create({
  uiContainer: {
    width: '75%',
  },
})

export default UIContainer
