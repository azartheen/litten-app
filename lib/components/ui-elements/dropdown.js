/**
 * @format
 * @flow
 */

import React from 'react'
import { Image, StyleSheet, Text, View } from 'react-native'
import {
  Menu,
  MenuOptions,
  MenuOption,
  MenuTrigger,
  renderers,
} from 'react-native-popup-menu'
import { iconChevron } from 'images'
import { rotate } from 'utils/functions'
import colors from 'styles/colors'

const { Popover } = renderers

const UIDropdown: (args: any) => React$Node = ({
  onSelect,
  options,
  placement = 'auto',
  selectedValue,
  style,
  ...otherProps
}) => {
  const translateSelectedValue = () =>
    options.find(({ value }) => value === selectedValue)?.label

  return (
    <View
      {...otherProps}
      style={StyleSheet.compose(styles.uiDropdownContainer, style)}>
      <Menu
        renderer={(props) => (
          <Popover
            {...props}
            placement={placement}
            anchorStyle={styles.anchorStyle}
          />
        )}
        onSelect={(value) => onSelect(value)}>
        <MenuTrigger>
          <View style={styles.uiDropdown}>
            <Text style={styles.selectedValueText}>
              {translateSelectedValue()}
            </Text>
            <Image
              source={iconChevron}
              resizeMode="contain"
              style={styles.iconChevron}
            />
          </View>
        </MenuTrigger>
        <MenuOptions
          customStyles={{ optionsContainer: styles.optionsContainer }}>
          {options.map(({ key, label, value }) => (
            <MenuOption key={key} value={value}>
              <Text
                style={
                  selectedValue === value
                    ? StyleSheet.compose(
                        styles.optionsText,
                        styles.optionsTextActive,
                      )
                    : styles.optionsText
                }>
                {label}
              </Text>
            </MenuOption>
          ))}
        </MenuOptions>
      </Menu>
    </View>
  )
}

const styles = StyleSheet.create({
  uiDropdownContainer: {
    display: 'flex',
    flexDirection: 'row',
    marginTop: 6,
    marginBottom: 6,
  },
  uiDropdown: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    padding: 12,
    borderRadius: 20,
    backgroundColor: colors.white,
  },
  selectedValueText: {
    fontWeight: '600',
    color: colors.blue,
  },
  iconChevron: {
    maxHeight: 12,
    maxWidth: 12,
    transform: [rotate('90deg', 'Z')],
  },
  optionsContainer: {
    padding: 10,
    borderRadius: 20,
    borderWidth: 0,
    shadowRadius: 0,
    shadowOffset: {
      height: 0,
      width: 0,
    },
    backgroundColor: colors.white,
  },
  optionsText: {
    padding: 2,
    fontWeight: '600',
    color: colors.darkGray,
  },
  optionsTextActive: {
    color: colors.blue,
  },
  anchorStyle: {
    height: 0,
  },
})

export default UIDropdown
