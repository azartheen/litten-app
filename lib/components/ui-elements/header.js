/**
 * @format
 * @flow
 */

import React from 'react'
import { StyleSheet, Text } from 'react-native'
import colors from 'styles/colors'

const UIHeader: (args: any) => React$Node = ({
  children,
  style,
  subheader = false,
  ...otherProps
}) => (
  <Text
    {...otherProps}
    style={[
      StyleSheet.compose(styles.uiHeader, style),
      subheader ? styles.uiSubHeader : undefined,
    ]}>
    {children}
  </Text>
)

const styles = StyleSheet.create({
  uiHeader: {
    fontSize: 24,
    fontWeight: '600',
    color: colors.black,
    paddingTop: 2,
    paddingBottom: 2,
    marginTop: 2,
    marginBottom: 2,
  },
  uiSubHeader: {
    fontSize: 16,
  },
})

export default UIHeader
