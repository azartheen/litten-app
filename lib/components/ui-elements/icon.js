/**
 * @format
 * @flow
 */

import React from 'react'
import { Image, StyleSheet, View } from 'react-native'
import colors from 'styles/colors'
import { elevated as elevatedStyle } from 'styles/common'
import { UI_ICON_SIZE_SMALL } from 'utils/constants'

const UIText: (args: any) => React$Node = ({
  circle = false,
  elevated = false,
  icon,
  iconStyle,
  style,
  ...otherProps
}) => (
  <View
    {...otherProps}
    style={[
      StyleSheet.compose(styles.uiIconContainer, style),
      circle ? styles.uiIconContainerCircle : undefined,
      elevated ? elevatedStyle : undefined,
    ]}>
    {icon && (
      <Image
        source={icon}
        resizeMode="contain"
        style={StyleSheet.compose(styles.uiIcon, iconStyle)}
      />
    )}
  </View>
)

const styles = StyleSheet.create({
  uiIconContainer: {
    height: UI_ICON_SIZE_SMALL,
    aspectRatio: 1,
    borderRadius: 4,
    padding: 8,
    backgroundColor: colors.white,
  },
  uiIconContainerCircle: {
    borderRadius: UI_ICON_SIZE_SMALL / 2,
  },
  uiIcon: {
    height: '100%',
    width: '100%',
  },
})

export default UIText
