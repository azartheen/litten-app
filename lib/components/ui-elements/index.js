import UIButton from './button'
import UIInput from './input'
import UILink from './link'
import UIListItem from './list-item'
import UIPasswordInput from './password-input'
import UIProgress from './progress'
import UISelect from './select'
import UIText from './text'

export {
  UIButton,
  UIInput,
  UILink,
  UIListItem,
  UIPasswordInput,
  UIProgress,
  UISelect,
  UIText,
}
