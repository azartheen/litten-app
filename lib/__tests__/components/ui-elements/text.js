import React from 'react'
import { UIText } from 'ui-elements'
import renderer from 'react-test-renderer'

describe('Snapshot test for the "UIText" component', () => {
  it('renders correctly', () => {
    const element = renderer.create(<UIText>Some text here</UIText>).toJSON()
    expect(element).toMatchSnapshot()
  })
})
