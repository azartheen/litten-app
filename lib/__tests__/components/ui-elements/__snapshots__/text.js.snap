// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for the "UIText" component renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#263238",
      "marginBottom": 8,
      "marginTop": 8,
      "paddingBottom": 6,
      "paddingTop": 6,
    }
  }
>
  Some text here
</Text>
`;
